hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/management/JpaControllerManagement.java

```diff
@@ -865,33 +867,35 @@ private void handleErrorOnAction(final JpaAction mergedAction, final JpaTarget m
      * @return a present controllerId in case the attributes needs to be
      *         requested.
      */
-    private Optional<String> handleFinishedAndStoreInTargetStatus(final JpaAction action) {
+    private Optional<String> handleFinishedAndStoreInTargetStatus(final long occurredAt, final JpaAction action) {
         final JpaTarget target = (JpaTarget) action.getTarget();
         action.setActive(false);
         action.setStatus(Status.FINISHED);
-        final JpaDistributionSet ds = (JpaDistributionSet) entityManager.merge(action.getDistributionSet());
-
-        target.setInstalledDistributionSet(ds);
-        target.setInstallationDate(System.currentTimeMillis());
+        if (target.getInstallationDate() == null || target.getInstallationDate() < occurredAt) {
```

Comment:
maybe target.getInstallationDate().longValue() will be more easy to understand the outboxing